enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverter.java

```diff
@@ -285,18 +285,27 @@ private static Optional<Attribute> buildAmountAttribute(final OperationTemplateP
         if (currency.isEmpty()) {
             return Optional.empty();
         }
-        final BigDecimal amountRaw;
+
+        BigDecimal amountRaw;
+        String amountFormatted;
+        String valueFormatted;
+        final String currencyRaw = currency.get();
+        final Locale locale = LocaleContextHolder.getLocale();
+
         try {
             amountRaw = new BigDecimal(amount.get());
-        } catch (NumberFormatException ex) {
-            logger.warn("Invalid number format: {}, skipping the AMOUNT attribute!", amount);
-            return Optional.empty();
+            amountFormatted = MonetaryConverter.formatAmount(amountRaw, currencyRaw, locale);
```

Comment:
Perhaps we can extract the logic which repeats in this pull request into a method? I realize this will require handling of inputs and inputs, however I am corned of future fixes which may update the logic only in one place. If this is not possible, leave it as is.

Reply:
Thanks for the comment, yes, it is possible.